salt/states/github.py

```diff
@@ -583,26 +598,134 @@ def repo_present(
 
     else:  # Repo does not exist - it will be created.
         if __opts__['test']:
-            ret['comment'] = 'Repo {0} is set to be created.'.format(name)
+            test_comments.append('Repo {0} is set to be created.'.format(name))
             ret['result'] = None
-            return ret
+        else:
+            add_params = dict(given_params)
+            add_params.update(kwargs)
+            result = __salt__['github.add_repo'](
+                name,
+                **add_params
+            )
+
+            if not result:
+                ret['result'] = False
+                ret['comment'] = 'Failed to create repo {0}.'.format(name)
+                return ret
+
+            # Turns out that trying to fetch teams for a new repo can 404 immediately
+            # after repo creation, so this waits until we can fetch teams successfully
+            # before continuing.
+            fetched = False
+            for attempt in range(3):
+                time.sleep(1)
+                try:
+                    __salt__['github.get_repo_teams'](
+                        name, profile=profile, **kwargs)
+                    fetched = True
+                    break
+                except CommandExecutionError as e:
+                    log.info("Attempt {0} to fetch new repo {1} failed".format(
+                        attempt, name))
+
+            if not fetched:
+                ret['result'] = False
+                ret['comment'] = 'Failed to verify repo {0} after creation.'.format(name)
+                return ret
 
-        add_params = dict(given_params)
-        add_params.update(kwargs)
-        result = __salt__['github.add_repo'](
-            name,
-            **add_params
-        )
-        if result:
             ret['changes']['repo'] = {
                 'old': None,
                 'new': 'Repo {0} has been created'.format(name)
             }
+
+    if teams is not None:
+        if __opts__['test'] and not target:
+            # Assume no teams if we're in test mode and the repo doesn't exist
+            current_teams = []
         else:
-            ret['result'] = False
-            ret['comment'] = 'Failed to create repo {0}.'.format(name)
-            return ret
+            current_teams = __salt__['github.get_repo_teams'](name, profile=profile)
+        current_team_names = set([t['name'] for t in current_teams])
 
+        # First remove any teams that aren't present
+        for team_name in current_team_names:
+            if team_name not in teams:
+                if __opts__['test']:
+                    test_comments.append('Repo {0} is set to be removed from team {1}.'
```

Comment:
Can we add this data to a `new`/`old` dictionary for team repos instead of only displaying comments when running with `test=True` here? It would look similar to the one you're generating below with `ret['changes'][team_name]` + the `old` and `new` keys, only we wouldn't actually delete the repos yet. That is only what it _would_ look like if the command is successful when running without test=true.